### ISSUE TITLE
develop: broken git-blame-micro-state

### ISSUE THREAD

**Issue by vbauerster** [syl20bnr/spacemacs#11568]
#### Description :octocat:
develop: broken git-blame-micro-state 

#### Reproduction guide :beetle:
- Start Emacs
- Press `space g b` in buffer under git

*Observed behaviour:* :eyes: :broken_heart:
Wrong type argument: commandp, magit-blame-addition

*Expected behaviour:* :heart: :smile:
git blame state

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 9da5da1d2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm emacs-lisp org multiple-cursors
      (git :variables git-magit-status-fullscreen t)
      (version-control :variables version-control-diff-side 'left)
      (auto-completion :variables auto-completion-return-key-behavior nil auto-completion-tab-key-behavior 'cycle auto-completion-enable-help-tooltip 'manual)
      (syntax-checking :variables syntax-checking-enable-by-default nil)
      (spell-checking :variables enable-flyspell-auto-completion t spell-checking-enable-by-default nil)
      html markdown treemacs yaml javascript lsp
      (go :variables go-backend 'lsp go-tab-width 4 go-format-before-save t gofmt-command "goimports" go-use-gometalinter t go-use-test-args "-race -timeout 10s" godoc-at-point-function 'godoc-gogetdoc)
      protobuf docker themes-megapack dash semantic
      (ranger :variables ranger-show-literal nil ranger-show-preview t ranger-show-hidden t ranger-cleanup-eagerly t ranger-cleanup-on-disable t ranger-ignored-extensions
              '("mkv" "flv" "iso" "mp4"))
      (shell :variables shell-default-shell 'eshell shell-default-height 30 shell-default-position 'bottom)
      shell-scripts
      (rust :variables rust-format-on-save t))
```
- System configuration features: JPEG RSVG IMAGEMAGICK NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS LCMS2

**Comment by sdwolfz** [comment]
I used to get this as well but it was fixed with the latest version of magit. If you still encounter this problem please turn on debugging `SPC t D` and post a stacktrace here. But I assume the problem will need to be reported upstream.

**Comment by vbauerster** [comment]
Here it is:
```
Debugger entered--Lisp error: (wrong-type-argument commandp magit-blame-addition)
  call-interactively(magit-blame-addition)
  (if (and (boundp 'magit-blame-mode) magit-blame-mode) nil (call-interactively 'magit-blame-addition))
  (let (golden-ratio-mode) (if (and (boundp 'magit-blame-mode) magit-blame-mode) nil (call-interactively 'magit-blame-addition)))
  spacemacs/git-blame-micro-state()
  funcall-interactively(spacemacs/git-blame-micro-state)
  call-interactively(spacemacs/git-blame-micro-state nil nil)
  command-execute(spacemacs/git-blame-micro-state)
```

**Comment by vbauerster** [comment]
I did some manual clean up, removed folders inside my `~/.emacs.d`:

- `.cache`
- `elpa`

Then after restarting emacs, git blame works.

**Comment by gone** [comment]
Deleted those folders and still seeing the issue :(

**Comment by felipernb** [comment]
I have the same issue running master. Cleaning cache and reinstalling packages didn't help.

**Comment by fkromer** [comment]
@vbauerster Your manual cleanup worked for me.

However it's a workaround, not a fix for the underlying problem. The problem pops up quite frequently and is really annoying. Is the root cause fixed in the `develop` branch with #11602 already?

**Comment by coventry** [comment]
In my case, this error was caused by a stale magit version. `M-x configuration-layer/update-packages` and restarting emacs fixed it for me.

Alternatively, you can still use `M-x magit-blame`
